src/dtgtk/thumbtable.c

```diff
@@ -2230,17 +2191,13 @@ static gboolean _accel_color(GtkAccelGroup *accel_group, GObject *acceleratable,
     if(id == darktable.develop->preview_pipe->output_imgid)
     {
       GList *res = dt_metadata_get(id, "Xmp.darktable.colorlabels", NULL);
-      res = g_list_first(res);
       gchar *result = NULL;
-      if(res != NULL)
+      for(GList *res_iter = res; res_iter; res_iter = g_list_next(res_iter))
       {
-        do
-        {
-          const GdkRGBA c = darktable.bauhaus->colorlabels[GPOINTER_TO_INT(res->data)];
-          result = dt_util_dstrcat(result,
-                                   "<span foreground='#%02x%02x%02x'>⬤ </span>",
-                                   (guint)(c.red*255), (guint)(c.green*255), (guint)(c.blue*255));
-        } while((res = g_list_next(res)) != NULL);
+        const GdkRGBA c = darktable.bauhaus->colorlabels[GPOINTER_TO_INT(res->data)];
```

Comment:
Smallest oversight. As a consequence, the tooltip for changing color labels shows all dots in the same color. :-)